tests/conftest.py

```diff
@@ -121,7 +119,7 @@ def update_nested_obj(id_path, obj, i):
                             # panda's deep copy is not recursive, so we have to do it
                             # again for nested objects
                             df[id_path[0]] = df[id_path[0]].map(
-                                lambda x: update_nested_obj(
+                                lambda x, i=i, id_path=id_path: update_nested_obj(
```

Comment:
:throws hands up: fine

Reply:
[another candidate](https://docs.astral.sh/ruff/rules/function-uses-loop-variable/#flake8-bugbear-b) for exclusion, but this is bugbear so maybe not: